Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/PropertyAssetCtrl.cpp

```diff
@@ -638,6 +638,21 @@ namespace AzToolsFramework
         {
             UpdateAssetDisplay();
         }
+        else if (!m_selectedAssetID.IsValid())
+        {
+            // If we get here without a valid assetId, it means the user has pressed the create button and we're waiting for an asset to be created.
+            // Use it if it's of the correct type.
+            AzFramework::AssetCatalogEventBus::Handler::BusDisconnect();
+
+            AZ::Data::AssetInfo assetInfo;
+            AZ::Data::AssetCatalogRequestBus::BroadcastResult(
+                assetInfo, &AZ::Data::AssetCatalogRequestBus::Events::GetAssetInfoById, assetId);
+
```

Comment:
Please consider slightly better performing lambda version of broadcast call?
```cpp
AZ::Data::AssetCatalogRequestBus::Broadcast([&](auto* interface) {
        assetInfo = interface->GetAssetInfoById(assetId);
    });
```

Reply:
Changed, although the compiler doesn't like auto in this case.

Reply:
Are you sure the issue is with the `auto` keyword? It could be the fact that MSVC compiler doesn't like a variable called `interface` as it sees it as a keyword.

Reply:
It's fine if I specify the type but still call it `interface`